test: Move `script_assets_tests` into its own suite

hebasto opened the pull request:
This PR ensures that the `script_assets_tests` test case is explicitly reported as "Skipped" when it is not run, making it clearer when running the test suite with `ctest`:

- on the master branch @ 9355578a77978a0c2f189bd7315a2883142d8119:
```
$ env -u DIR_UNIT_TEST_DATA ctest --test-dir build -j 16 -R "^script_"
Internal ctest changing into directory: /home/hebasto/git/bitcoin/build
Test project /home/hebasto/git/bitcoin/build
    Start 87: script_tests
    Start 83: script_p2sh_tests
    Start 85: script_segwit_tests
    Start 86: script_standard_tests
    Start 84: script_parse_tests
1/5 Test #84: script_parse_tests ...............   Passed    0.11 sec
2/5 Test #86: script_standard_tests ............   Passed    0.11 sec
3/5 Test #85: script_segwit_tests ..............   Passed    0.12 sec
4/5 Test #83: script_p2sh_tests ................   Passed    0.12 sec
5/5 Test #87: script_tests .....................   Passed    0.36 sec

100% tests passed, 0 tests failed out of 5

Total Test time (real) =   0.37 sec
```
- with this PR:
```
$ env -u DIR_UNIT_TEST_DATA ctest --test-dir build -j 16 -R "^script_"
Internal ctest changing into directory: /home/hebasto/git/bitcoin/build
Test project /home/hebasto/git/bitcoin/build
    Start 83: script_assets_tests
    Start 88: script_tests
    Start 84: script_p2sh_tests
    Start 86: script_segwit_tests
    Start 87: script_standard_tests
    Start 85: script_parse_tests
1/6 Test #85: script_parse_tests ...............   Passed    0.11 sec
2/6 Test #83: script_assets_tests ..............***Skipped   0.12 sec
3/6 Test #86: script_segwit_tests ..............   Passed    0.11 sec
4/6 Test #87: script_standard_tests ............   Passed    0.11 sec
5/6 Test #84: script_p2sh_tests ................   Passed    0.12 sec
6/6 Test #88: script_tests .....................   Passed    0.36 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) =   0.37 sec

The following tests did not run:
	 83 - script_assets_tests (Skipped)
$ env DIR_UNIT_TEST_DATA=/home/hebasto/git/bitcoin/qa-assets/unit_test_data ctest --test-dir build -j 16 -R "^script_"
Internal ctest changing into directory: /home/hebasto/git/bitcoin/build
Test project /home/hebasto/git/bitcoin/build
    Start 83: script_assets_tests
    Start 88: script_tests
    Start 84: script_p2sh_tests
    Start 86: script_segwit_tests
    Start 87: script_standard_tests
    Start 85: script_parse_tests
1/6 Test #85: script_parse_tests ...............   Passed    0.11 sec
2/6 Test #87: script_standard_tests ............   Passed    0.11 sec
3/6 Test #86: script_segwit_tests ..............   Passed    0.11 sec
4/6 Test #84: script_p2sh_tests ................   Passed    0.12 sec
5/6 Test #88: script_tests .....................   Passed    0.35 sec
6/6 Test #83: script_assets_tests ..............   Passed    1.58 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) =   1.58 sec
```

hebasto commented:
Friendly ping @maflcko ;)

yuvicc reviewed (commented):
tACK 0859fc22f0baf00259cf964f9d39d36fba6cd291

- making skipped tests explicitly appear as 'Skipped' rather than silently omitting

```
    Start 89: script_tests
    Start 85: script_p2sh_tests
    Start 87: script_segwit_tests
    Start 88: script_standard_tests
    Start 86: script_parse_tests
    Start 84: script_assets_tests
1/6 Test #86: script_parse_tests ...............   Passed    0.16 sec
2/6 Test #84: script_assets_tests ..............***Skipped   0.16 sec
3/6 Test #88: script_standard_tests ............   Passed    0.17 sec
4/6 Test #87: script_segwit_tests ..............   Passed    0.17 sec
5/6 Test #85: script_p2sh_tests ................   Passed    0.17 sec
6/6 Test #89: script_tests .....................   Passed    0.37 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) =   0.38 sec

The following tests did not run:
	 84 - script_assets_tests (Skipped)
```

maflcko reviewed (approved):
makes sense to be explicit here to avoid confusion

lgtm ACK 0859fc22f0baf00259cf964f9d39d36fba6cd291 🐽

<details><summary>Show signature</summary>

Signature:

```
untrusted comment: signature from minisign secret key on empty file; verify via: minisign -Vm "${path_to_any_empty_file}" -P RWTRmVTMeKV5noAMqVlsMugDDCyyTSbA3Re5AkUrhvLVln0tSaFWglOw -x "${path_to_this_whole_four_line_signature_blob}"
RUTRmVTMeKV5npGrKx1nqXCw5zeVHdtdYURB/KlyA/LMFgpNCs+SkW9a8N95d+U4AP1RJMi+krxU1A3Yux4bpwZNLvVBKy0wLgM=
trusted comment: lgtm ACK 0859fc22f0baf00259cf964f9d39d36fba6cd291 🐽
oguvM7vjti8dmtabMmgzCKuxp+/7OOHqZVEMvjj+RjTVxfTJLoC+HJhdnAQ11Y5a2tsJI5smtPebrAKwdUAKBA==
```

</details>

hebasto committed:
test: Move `script_assets_tests` into its own suite

This change ensures that the `script_assets_tests` test case is
explicitly reported as "Skipped" when it is not run, making it clearer
when running the test suite with `ctest`.

hebasto commented on src/test/script_assets_tests.cpp:
```diff
@@ -0,0 +1,189 @@
+BOOST_AUTO_TEST_SUITE(script_assets_tests)
+
+/* Wrapper around ProduceSignature to combine two scriptsigs */
+SignatureData CombineSignatures(const CTxOut& txout, const CMutableTransaction& tx, const SignatureData& scriptSig1, const SignatureData& scriptSig2)
```
Thanks! Dropped.

ajtowns commented:
ACK c40dbbbf77078bb86a652ca151b472e7bef61ae0

You could consider rebasing it on top of #32945 in order to not move the code to a different file, but that still involves moving code about and only avoids duplicating the `ToScript` function across both files.

achow101 commented:
ACK c40dbbbf77078bb86a652ca151b472e7bef61ae0

l0rinc commented:
test-only post-merge ACK

It's reporting `script_assets_tests` as skipped on Mac

<details>
<summary>Details</summary>

```bash
129/144 Test  #86: script_assets_tests ..................***Skipped   0.36 sec
...
The following tests did not run:
         86 - script_assets_tests (Skipped)
```

</details>

 and on Linux

<details>
<summary>Details</summary>

```bash
139/144 Test  #86: script_assets_tests ..................***Skipped   0.20 sec
...
The following tests did not run:
         86 - script_assets_tests (Skipped)
```

</details>

While [CI](https://github.com/bitcoin/bitcoin/actions/runs/15775500901/job/44468932743?pr=31576) ran it successfully

<details>
<summary>Details</summary>

```bash
131/143 Test  #87: script_assets_tests ..................   Passed  184.12 sec
```

</details>
